### ISSUE TITLE
IntelliSense Issues with C++ Modules hints

### ISSUE THREAD

**Issue by xBLACKICEx** [microsoft/vscode-cpptools#12985]
### Environment

- OS and Version: Win11 24H2 21600.2448
- VS Code Version: 1.95.2
- C/C++ Extension Version: v1.23.1 (pre-release)

### Bug Summary and Steps to Reproduce

Bug Summary: 

with C++ modules:
**IntelliSense fails to fetch function parameter names.**
![Image](https://github.com/user-attachments/assets/66af4c15-38ca-4b44-abf2-3ea3ffc2fca4)

**Inline hints for parameter names are not displayed.**
![Image](https://github.com/user-attachments/assets/7f743db7-6634-453a-b351-5f302a75acad)

**Type deduction is overly verbose for certain templates (e.g., std::basic_string instead of std::string).**
![Image](https://github.com/user-attachments/assets/cb2c91ae-cc2c-4d77-b4a3-8625582e3628)

---
**Expected behavior:**
![Image](https://github.com/user-attachments/assets/ae45ffb9-0d7d-4e17-8319-1d853d9741c5)

### Configuration and Logs

```shell
{
    "configurations": [
        {
            "name": "Win32",
            "includePath": [
                "${workspaceFolder}/**"
            ],
            "cStandard": "c23",
            "cppStandard": "c++23",
            "intelliSenseMode": "windows-msvc-x64",
            "compilerPath": "cl.exe",
            "compilerArgs": [
                "/ifcSearchDir",
                "${workspaceFolder}/ifc"
            ]
        }
    ],
    "version": 4
}

log: no log found at output (C/C++ Task was not displayed)
```

**Comment by sean-mcmanus** [comment]
@xBLACKICEx This is a duplicate of https://developercommunity.visualstudio.com/t/C:-Pop-up-tip-shows-incorrect-signatur/10405046 -- which they closed as Lower Priority -- but it seems like a relatively major issue to me for users using modules.

**Comment by bobbrow** [comment]
@sean-mcmanus I agree with you on this. Can you open a new ticket? I didn't recall seeing this issue and it looked like Russell had triaged it before I saw it.
